### PR TITLE
Add Material 3 Popup Menu example and update existing example

### PR DESCRIPTION
**Pull request by TahaTesser**
Material 3 Popup Menu was added in https://github.com/flutter/flutter/pull/103606
Part of https://github.com/flutter/flutter/issues/101048

Screenshot 2022-10-28 at 17 42 04

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/material/popup_menu/popup_menu.0.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Flutter code sample for [PopupMenuButton].
+
+import 'package:flutter/material.dart';
+
+// This is the type used by the popup menu below.
```
Comment (gspencergoog) [inline examples/api/lib/material/popup_menu/popup_menu.0.dart:9]:
```suggestion
// This is the value type used by the popup menu below.
```

File: examples/api/lib/material/popup_menu/popup_menu.1.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Flutter code sample for [PopupMenuButton].
+
+import 'package:flutter/material.dart';
+
+// This is the type used by the popup menu below.
```
Comment (gspencergoog) [inline examples/api/lib/material/popup_menu/popup_menu.1.dart:9]:
```suggestion
// This is the value type used by the popup menu below.
```

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 3]
```diff
@@ -1017,10 +1017,17 @@ typedef PopupMenuItemBuilder<T> = List<PopupMenuEntry<T>> Function(BuildContext
 /// platform).
 ///
 /// {@tool dartpad}
-/// This example shows a menu with four items, selecting between an enum's
-/// values and setting a `_selectedMenu` field based on the selection
+/// This example shows a menu with three items, selecting between an enum's
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/popup_menu.dart:1020]:
```suggestion
/// This example shows a menu that selects among three items whose values are determined by an enum.
```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Thanks so much for making all these examples: our users really love them.

**Comment by TahaTesser** [comment]
@gspencergoog 
For some reason the google test is failing, rebasing doesn't fix the issue.

**Comment by gspencergoog** [comment]
Fixed it.

**Comment by TahaTesser** [comment]
> Fixed it.

Thank you

**Pull request merged**